Adding translation to Using Typescript (#872)
reactjs/es.react.dev: src/content/learn/typescript.md

---
title: Usar TypeScript
re: https://github.com/reactjs/react.dev/issues/5960
---

<Intro>

> TypeScript es una opción popular para incorporar definiciones de tipos en bases de código de JavaScript. De manera nativa, TypeScript [es compatible con JSX](/learn/writing-markup-with-jsx) y puedes obtener soporte completo para React en la web al añadir [`@types/react`](https://www.npmjs.com/package/@types/react) y [`@types/react-dom`](https://www.npmjs.com/package/@types/react-dom) a tu proyecto.

Reviewer: ```suggestion
TypeScript es una forma popular de añadir definiciones de tipos a bases de código JavaScript. TypeScript [soporta JSX](/learn/writing-markup-with-jsx) y puedes obtener soporte completo para React Web añadiendo [`@types/react`](https://www.npmjs.com/package/@types/react) y [`@types/react-dom`](https://www.npmjs.com/package/@types/react-dom) a tu proyecto.
```

</Sandpack>

Existen muchos tipos de eventos disponibles en los tipos de React - la lista completa se puede encontrar [aquí](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b580df54c0819ec9df62b0835a315dd48b8594a9/types/react/index.d.ts#L1247C1-L1373), la cual se basa en los [eventos más populares del DOM](https://developer.mozilla.org/en-US/docs/Web/Events).

Al determinar el tipo que estás buscando, puedes mirar primero la información emergente para el controlador de eventos que estás utilizando, lo cual mostrará el tipo del evento.

Si necesitas utilizar un evento que no está incluido en esta lista, puedes emplear el tipo `React.SyntheticEvent`, que es el tipo base para todos los eventos.

### Elementos hijos {/*typing-children*/}

> Hay dos enfoques comunes para describir los elementos hijos de un componente. El primero es usar el tipo `React.ReactNode`, que es una unión de todos los tipos posibles que se pueden pasar como elementos hijos en JSX:

Reviewer: ```suggestion
Hay dos caminos comunes para describir los hijos de un componente. El primero es utilizar el tipo `React.ReactNode`, que es una unión de todos los tipos posibles que se pueden pasar como hijos en JSX:
```

}
```

> Esta es una definición bastante amplia de children. La segunda opción es emplear el tipo `React.ReactElement`, que incluye solamente elementos JSX y no incluye primitivos de JavaScript como strings o números:

Reviewer: ```suggestion
Esta es una definición muy amplia de hijos. El segundo es utilizar el tipo `React.ReactElement`, que es sólo elementos JSX y no primitivas JavaScript como strings o numbers:
```

}
```

> Ten en cuenta que no puedes utilizar TypeScript para describir que los elementos hijos sean de un cierto tipo de elementos JSX, por lo que no puedes usar el sistema de tipos para describir un componente que solo acepta elementos `<li>` como hijos. 

Reviewer: ```suggestion
Ten en cuenta que no puedes usar TypeScript para describir que los hijos son un cierto tipo de elementos JSX, por lo que no puedes usar el sistema de tipos para describir un componente que sólo acepta hijos `<li>`.
```
